### ISSUE TITLE
Uncommenting "opacity-rule" in sample config breaks compton

### ISSUE THREAD

**Issue by godenji** [chjj/compton#289]
Get an "Unexpected expression" error when uncommenting sample config line:

`opacity-rule = [ "80:class_g = 'URxvt'" ];`

Running Compton 0.1.0-1on Fedora 21 (kernel 4.0.4-201) with i3 4.10.2.

While the Compton effects are impressive, the feature I'm most interested is transparent terminals; I can't (AFAICT) achieve this without being able to apply an opacity along with the rule.

Ideas/clues appreciated.

Thanks

**Comment by skry** [comment]
Putting "URxvt.background : [85]black" or something similar to your ~/.Xresources is not an option?

**Comment by godenji** [comment]
@skry thanks, I'm not actually using urxvt (run Gnome terminal).

In the end I'd like to have transparency *only* for terminal windows, so assume that `opacity-rule` is the config option to use.

**Comment by skry** [comment]
@godenji I'm not familiar with Gnome Terminal but most if not all have an option for transparency. Anyway, yes opacity-rule would be the option to use if your term lacks the option. I'm running version 0.1_beta2 on Arch. I copy pasted the opacity-rule line from your first post and couldn't reproduce the issue. No errors and opacity seems to be set accordingly. You sure your config is otherwise OK?

**Comment by godenji** [comment]
@skry the issue is independent of whatever I put in as the matching window class; just copy pasted the sample config, uncommented the `opacity-rule` line, and get error above.

Must be something with my setup as Compton otherwise works fine (amazingly well in fact) with the sample config and offending line left commented.

And yes, gnome terminal does have built-in transparency support, I'm just trying to get Compton to run the effects show ;-)

**Comment by skry** [comment]
@godenji Strange, can't really say nothing about what could be the problem then. If it is of any consolation, compositor handles the transparency most likely anyway. So feel free to set the alpha from Gnome Terminal as the end result is the same which is compton doing its job :)

**Comment by godenji** [comment]
@skry right, I'll wind up doing it via gnome terminal, but it seemed like an "issue" to me so went ahead and posted one ;-)

**Comment by richardgv** [comment]
Sorry for the late reply. This is a known issue that was fixed in 796e2c6f, right before `0.1_beta2` was released, so probably your version of compton is a bit old.

By the way, using GNOME terminal itself for transparency is generally preferred, as it uses ARGB background for the purpose, instead of applying opacity to the whole window, so the foreground, titlebar, and other GUI elements will be fully opaque.

_The issue will be closed if there is no reply in 2 weeks._
